package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -820,7 +825,8 @@ def _selection_serial(self, universe, selection_str):
         selection = []
         for ts in universe.trajectory:
             selection.append(universe.select_atoms(selection_str))
-            print(ts.frame)
+            sys.stdout.write("\rSelecting atoms from frame: %i"%ts.frame)
```

Comment:
ProgressMeter is made for this use case. Bare prints are not good because you cannot switch them off easily (short of redirecting stdout).